src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandlerBuilderFilter.cs

```diff
@@ -11,16 +13,17 @@ namespace Microsoft.Extensions.Http
     // Internal so we can change the requirements without breaking changes.
     internal sealed class LoggingHttpMessageHandlerBuilderFilter : IHttpMessageHandlerBuilderFilter
     {
-        private readonly ILoggerFactory _loggerFactory;
+        private readonly Lazy<ILoggerFactory> _loggerFactory;
         private readonly IOptionsMonitor<HttpClientFactoryOptions> _optionsMonitor;
 
-        public LoggingHttpMessageHandlerBuilderFilter(ILoggerFactory loggerFactory, IOptionsMonitor<HttpClientFactoryOptions> optionsMonitor)
+        public LoggingHttpMessageHandlerBuilderFilter(IServiceProvider serviceProvider, IOptionsMonitor<HttpClientFactoryOptions> optionsMonitor)
         {
-            ThrowHelper.ThrowIfNull(loggerFactory);
+            ThrowHelper.ThrowIfNull(serviceProvider);
             ThrowHelper.ThrowIfNull(optionsMonitor);
 
-            _loggerFactory = loggerFactory;
             _optionsMonitor = optionsMonitor;
+
+            _loggerFactory = new Lazy<ILoggerFactory>(serviceProvider.GetRequiredService<ILoggerFactory>, LazyThreadSafetyMode.PublicationOnly);
```

Comment:
Add a comment explaining why.

Reply:
Why PublicationOnly in this lazy and ExecutionAndPublication in DefaultHttpClientFactory lazy?

Reply:
I guess we don't need to guard this one at all, given that DI should already guard creation of a singleton.

DefaultHttpClientFactory lazy is creating ILogger, which is most possibly a new instance every time CreateLogger is invoked.